capsules/extra/src/hc_sr04.rs

```diff
@@ -0,0 +1,158 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+use core::cell::Cell;
+
+use kernel::hil::gpio::{Client, InterruptEdge};
+use kernel::hil::sensors::{self, Distance, DistanceClient};
+use kernel::hil::time::{Alarm, Ticks};
+use kernel::hil::time::{AlarmClient, ConvertTicks};
+use kernel::utilities::cells::OptionalCell;
+use kernel::ErrorCode;
+
+/// Maximum duration for the echo pulse to be measured in milliseconds.
+// As specified in the datasheet:
+// https://www.handsontec.com/dataspecs/HC-SR04-Ultrasonic.pdf,
+// the maximum time for the echo pulse to return is around 38 milliseconds
+// for a maximum distance of approximately 4 meters. We use a slightly
```

Comment:
Something doesn't math right here.

`343 m/s * 0.05 seconds = 17.15 meters` (38 ms -> 13 meters)

That's probably halved, since you need to transmit and receive the echo, but still doesn't work out to 4 meters...

Reply:
For 4 meters, the echo time would be about 23 ms under standard temperature and pressure conditions. However, because this applies only at those specific conditions, I used 38 ms to adapt for variations in real-world environments.